### PR TITLE
Outbound queue

### PR DESCRIPTION
**Pull request by SCdF**
Allows for configuring certain documents to be mapped into an outbound
structure and pushed to external web services.

medic/medic#5499

### CODE REVIEW

File: sentinel/src/schedule/outbound.js [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+          password: authConf.password
+        },
+        // TODO: I think Request adds this for us?
+        // headers: {
+        //   'Content-Type': 'multipart/form-data'
+        // },
+        url: conf.destination.base_url + authConf.path
+      };
+
+      return request(authOptions)
```
Comment (kennsippell) [inline sentinel/src/schedule/outbound.js]: Have you considered using the `async` pattern when working with promises in new code in api and sentinel?

Comment (SCdF) [reply sentinel/src/schedule/outbound.js]: I **completely forgot** we can use that now!

We _can_ use that now right?

If so I'll rewrite, it's a good self-contained situation to play with it.

Comment (SCdF) [reply sentinel/src/schedule/outbound.js]: I forgot to write an update here: we don't support async / await without also changing our eslint config. Which is no problem, but I have too much on my plate to justify rewriting existing working code just to change patterns, so I'll make that change next time and write async from scratch

**Comment by SCdF** [comment]
Please review @kennsippell.

Note that it is not based off master, but instead of a PR of Diana's. Once her PR is AT'd and merged I'll have to drop and re-create this PR I think, but we might as well get it reviewed now.

File: api/package.json [hunk 2]
```diff
@@ -24,9 +24,9 @@
     "@medic/server-checks": "^1.0.1",
     "@medic/task-utils": "^1.0.0",
     "@medic/tombstone-utils": "^1.0.0",
-    "medic-transitions-beta": "^0.0.5",
     "@medic/translation-utils": "^1.0.1",
     "@medic/view-map-utils": "^1.0.0",
+    "@scdf/transitions": "^1.1.0",
```
Comment (SCdF) [inline api/package.json]: We do this so we can publish the packages before doing the actual merge.

Once this is ready to roll I'll go through and create a proper PR that uses the medic org etc

**Comment by SCdF** [comment]
@kennsippell don't worry about this yet, I want to do a couple more things here. I'll ping you when I'm done

**Pull request closed without merging**